wicker/core/column_files.py

```diff
@@ -178,15 +178,70 @@ def _write_column(self, column_name: str) -> None:
         self.write_buffers[column_name] = self._get_new_buffer()
 
 
-class ColumnBytesFileCache:
+class ColumnBytesFileReader:
+    """A reader class for column bytes files to inherit upon for caches and cacheless"""
+
+    def __init__(
+        self,
+        dataset_name: Optional[str] = None,
+        path_factory: WickerPathFactory = S3PathFactory(),
+    ) -> None:
+        self._column_bytes_root_path = path_factory._get_column_concatenated_bytes_files_path(dataset_name=dataset_name)
+        self._dataset_name = dataset_name
+        self._path_factory = path_factory
+
+    def _read_column_bytes_file(
+        self, column_bytes_file_info: ColumnBytesFileLocationV1, column_bytes_file_path: str
+    ) -> bytes:
+        """Read a column bytes file in a standard location.
+
+        Reads the column bytes file from the column bytes info class.
+
+        Args:
+            column_bytes_file_info (ColumnBytesFileLocationV1): Location info for a ColumnBytes
+            file.
+            column_bytes_file_path (str): Path where the ColumnBytes file is written from which to read.
+
+        Returns:
+            bytes: File bytes corresponding to column bytes data.
+        """
+        with open(column_bytes_file_path, "rb") as f:
+            f.seek(column_bytes_file_info.byte_offset)
+            return f.read(column_bytes_file_info.data_size)
+
+    def read(self, column_bytes_file_info: ColumnBytesFileLocationV1) -> bytes:
+        """Read data from column bytes file.
+
+        :param column_bytes_file_info: Location object to column bytes file on data store
+        :type column_bytes_file_info: ColumnBytesFileLocationV1
+        :return: bytes read from the column bytes file
+        """
+        column_concatenated_bytes_file_path = os.path.join(
+            self._column_bytes_root_path, str(column_bytes_file_info.file_id)
+        )
+
+        return self._read_column_bytes_file(
+            column_bytes_file_info=column_bytes_file_info, column_bytes_file_path=column_concatenated_bytes_file_path
+        )
+
+    def resolve_pointers(
+        self,
+        example: validation.AvroRecord,
+        schema: schema.DatasetSchema,
+    ):
+        visitor = ResolvePointersVisitor(self, example, schema)
+        return visitor.resolve_pointers()
+
+
+class ColumnBytesFileCache(ColumnBytesFileReader):
     """A read-through caching abstraction for accessing ColumnBytesFiles"""
 
     def __init__(
         self,
         local_cache_path_prefix: str = "/tmp",
         filelock_timeout_seconds: int = -1,
```

Comment:
Below since the `path_factory` and `storage` might not be S3-specific in the future, let's expand their types to be `WickerPathFactory` and `AbstractDataStorage` instead.

Reply:
> Below since the `path_factory` and `storage` might not be S3-specific in the future, let's expand their types to be `WickerPathFactory` and `AbstractDataStorage` instead.

Thanks!